pkg/solana/config/config.go

```diff
@@ -21,6 +21,7 @@ var defaultConfigSet = configSet{
 	TxConfirmTimeout:    15 * time.Second,       // duration before discarding tx as unconfirmed
 	SkipPreflight:       true,                   // to enable or disable preflight checks
 	Commitment:          rpc.CommitmentConfirmed,
+	MaxRetries:          nil, // max number of retries, when nil - rpc node will do a reasonable number of retries
```

Comment:
can we have this default to `0`? since the sendTx function is primarily used in the context of the transaction manager and that implements custom retry logic

Reply:
Yeah, having it `0` by default would mean we won't be able to set it to `nil` and thus no way to let the rpc node decide  on the number of retries (default rpc node functionality). I would prefer setting `0` explicitly in config since we change the default rpc node's behaviour.

Reply:
But in case it's somehow difficult to set up the config, let's have it `0`, it's one-liner anyways

Reply:
we could introduce an function (like `SendTxWithOpts`) where `opts` can be passed in? if that is nil, then it uses the default set (where we can leave `maxRetries` as nil, sort of like how solana-go manages it)

Reply:
Alright, I can do it this way! Is there place somewhere in the doc where I can describe it?

Reply:
for now, i would say leave some comments on the function...i just looked at files in the docs folder and they are out of date (they are from before the nodes/chains refactor, I will update them once we merge this PR)